### PR TITLE
Issue #43 - Support benchmarks with subbenchmarks

### PR DESCRIPTION
**Pull request by armenzg**
We are adding a new route for Quantum that allows showing a breakdown
for each subbenchmarks for a given benchmark.

You can test it by visiting https://http://health.graphics/quantum/linux64/ARES6

### CODE REVIEW

**Comment by armenzg** [comment]
Hi @wlach do you have some feedback on this?

CC @stuartphilp @jmaher feedback wanted

The feature works like this:

- Visit https://platform-health-pr-62.herokuapp.com/quantum
- Go down to the "#3 Other benchmarks" section
- Click on the "Breakdown" link [1]
  - **TODO**: I have to adjust the CSS there to move the text to the right
- This will take you to the [breakdown view](https://platform-health-pr-62.herokuapp.com/quantum/linux64/ARES6) [2]

I've also added bookmarkibility:

- For instance https://platform-health-pr-62.herokuapp.com/quantum/linux64/ARES6#ML_firstIteration
  -  **TODO**: I've noticed the icon does not work; I will fix that

[1] 
image

[2]
image

**Comment by jmaher** [comment]
the linking to a specific test doesn't work for me, but the rest does.

some general feedback:
1) I don't like the curvy lines- it makes it harder for me to see longer
term trends
2) we had a spike on march 6th- that adjusted all the Y axis scale- can we
find a way to set the scale on 95% of the data and mouse over for the
outliers?
3) I would like geomean to be at the top- that is the benchmark score

Otherwise- the flow from the platform health to the benchmark looks good.

File: src/quantum/subbenchmarks.jsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+                    <a href={el.meta.url} target="_blank">{el.meta.test}</a>
+                  </div>
+                  <MetricsGraphics
+                    key={el.meta.test}
+                    data={[el.data]}
+                    {...{
+                      x_accessor: 'datetime',
+                      y_accessor: 'value',
+                      min_y_from_data: true,
+                      full_width: true,
```
Comment (wlach) [inline src/quantum/subbenchmarks.jsx]: I don't think the chart should be an area chart. Try setting `area: false` here.

I believe that should also take care of the curving issue that jmaher mentioned

**Comment by wlach** [comment]
@armenzg check that, I think the weird curviness is due to interpolation. Try using a linear curvature instead:

https://github.com/mozilla/missioncontrol/blob/4c893c255c8f3dd623ff380c4c111b8980f8de94/frontend/ui/detailgraph.jsx#L25

**Comment by armenzg** [comment]
Thanks for the suggestion!
I don't think it is much of an improvement.

I will try with just dots to see how it looks.

image

**Comment by wlach** [comment]
That actually looks a lot better to me. But points or some kind of sliding confidence window may be even better. I guess you can experiment with what looks good to people.

**Comment by armenzg** [comment]
I'm satisfied with what I have.

This is what I will be merging:
* Support 2nd title for Widget criteria
  * Cleaned up CSS issues for Widget criteria (See issue #45 )
* Support for subbenchmarks API ([/quantum/linux64/ARES6](https://platform-health-pr-73.herokuapp.com/quantum/linux64/ARES6))

I'm leaving some things out of this PR:
* #70 - Enable section with new benchmarks
  * Since we don't care to show Linux64 and it was only until we've Windows support
  * Enabled Linux64 benchmarks in the `enable_linux64_benchmarks` branch
  * You can see the deployment [here](https://platform-health-pr-73.herokuapp.com/quantum) 
* #71 - Confidence band

**Pull request merged**